src/main/scala/beam/agentsim/agents/ridehail/RideHailManager.scala

```diff
@@ -947,6 +967,39 @@ class RideHailManager(
     }
   }
 
+  def throwRideHailFleetStateEvent(tick: Int): Unit = {
+    val tick = modifyPassengerScheduleManager.getCurrentTick.get
+    val inServiceRideHailVehicles = vehicleManager.inServiceRideHailVehicles.values.toList
+    val inServiceRideHailStateEvents = calculateCavEvs(inServiceRideHailVehicles, "InService", tick)
+    eventsManager.processEvent(inServiceRideHailStateEvents)
+    val outOfServiceRideHailVehicles = vehicleManager.outOfServiceRideHailVehicles.values.toList
+    val outOfServiceRideHailStateEvents = calculateCavEvs(outOfServiceRideHailVehicles, "offline", tick)
+    eventsManager.processEvent(outOfServiceRideHailStateEvents)
+    val idleRideHailEvents = vehicleManager.idleRideHailVehicles.values.toList
+    val idleRideHailStateEvents = calculateCavEvs(idleRideHailEvents, "idle", tick)
+    eventsManager.processEvent(idleRideHailStateEvents)
+  }
+
+  def calculateCavEvs(
+    rideHailAgentLocations: List[RideHailAgentLocation],
+    vehicleType: String,
+    tick: Int
+  ): RideHailFleetStateEvent = {
```

Comment:
nitpick: you can change `rideHailAgentLocations` to be something like `Iterable` then you don't need to convert it to list on https://github.com/LBNL-UCB-STI/beam/pull/2269/files#diff-168d2e3aaa40588af77cab0acf151676R978

Reply:
done